pycue/opencue/cloud/gce_api.py

```diff
@@ -0,0 +1,161 @@
+
+        self.target_size = self.data["targetSize"]
+
+        if self.current_instances_size == self.target_size:
+            return self.current_instances_size
+        else:
+            return "{current_size} -> {target_size}".format(current_size=self.current_instances_size,
+                                                            target_size=self.target_size)
+
+    def resize(self, size=None):
+        request = self.connection_manager.service.instanceGroupManagers().resize(
+            project=self.connection_manager.project, zone=self.connection_manager.zone,
+            instanceGroupManager=self.name, size=size)
+        response = request.execute()
+
+    def status(self):
+        """
+        Use the info gained from group size info to customize status column
+        :return:
+        """
+        if self.data["status"].get("isStable"):
+            return "STABLE"
+        else:
+            if self.target_size > self.current_instances_size:
+                return "BUSY: SCALING UP"
+            elif self.target_size < self.current_instances_size:
+                return "BUSY: SCALING DOWN"
+
+            return "BUSY: IN OPERATION"
+
+    def id(self):
+        return self.data["id"]
+
+
+class GoogleCloudManager(opencue.cloud.api.CloudManager):
+
+    def __init__(self):
+        super(GoogleCloudManager, self).__init__()
+        # TODO : to be replaced with a better way to handle authentication
+        self.project = 'gsoc-opencue-test-bed'
+        self.zone = 'us-central1-a'
+        self.credentials = None
+        self.service = None
+
+    def signature(self):
+        return "google"
+
+    def connect(self):
+        """
+        Connect to the GCE : For now with application defaults
+        :return:
+        """
+        self.credentials = oauth2client.client.GoogleCredentials.get_application_default()
+        self.service = googleapiclient.discovery.build('compute', 'v1', credentials=self.credentials)
+
+    def get_all_groups(self):
+        cigs = []
+        request = self.service.instanceGroupManagers().list(project=self.project, zone=self.zone)
+        while request is not None:
+            response = request.execute()
+            for instance_group_manager in response['items']:
+                new_cig = GoogleCloudGroup(data=instance_group_manager, connection_manager=self)
+                # Call get_instances to update the actual
+                # number of instances running for the group
+                new_cig.get_instances()
+                cigs.append(new_cig)
+            request = self.service.instanceGroupManagers().list_next(previous_request=request,
+                                                                     previous_response=response)
+
+        return cigs
+
+    @opencue.cloud.gce_exception_util.googleRequestExceptionParser
+    def create_managed_group(self, name, size, template):
+        # TODO : Use request ID to handle multiple create button clicks
```

Comment:
Elsewhere in CueGUI this is handled at the plugin level -- a creation request gets added to some internal cache within the widget, then the widget's tick method iterates over the requests and takes action on each of them. In that logic you could do things like de-duping requests as well.

Doesn't need to happen in this PR, just wanted to note for later.